src/logic/basic.lean

```diff
@@ -148,6 +148,9 @@ lemma ne_comm {α} {a b : α} : a ≠ b ↔ b ≠ a := ⟨ne.symm, ne.symm⟩
   (∀ {c}, a = c ↔ b = c) ↔ (a = b) :=
 ⟨λ h, by rw h, λ h a, by rw h⟩
 
+/-- Gadget to help port instances from definitionally equal types -/
+def infer_instance_as (cl : Sort*) [I : cl] : cl := I
```

Comment:
I don't think this actually works. You can't have typeclass parameters where the typeclass itself is a variable. I'm not sure how you intend to use it, but this doesn't work:
```lean
def infer_instance_as (cl : Sort*) [I : cl] : cl := I

def foo := nat

instance : linear_order foo := infer_instance_as (linear_order nat)
-- failed to synthesize type class instance for
-- ⊢ linear_order foo
```

Reply:
That's how I remember using it but it doesn't seem to work. It's not needed anymore anyway so the point is moot.